lightning/src/ln/onion_utils.rs

```diff
@@ -169,7 +169,12 @@ pub(super) fn build_onion_payloads(path: &Vec<RouteHop>, total_msat: u64, paymen
 							payment_secret: payment_secret.clone(),
 							total_msat,
 						})
-					} else { None },
+					} else if keysend_preimage.is_some() && value_msat < total_msat {
+						Some(msgs::FinalOnionHopData {
+							payment_secret: PaymentSecret([0u8; 32]),
+							total_msat,
```

Comment:
Not super sure what's the best way to fill payment_secret here, so suggestions welcome :)

Reply:
Changed this to set payment secret to be the preimage